index.js

```diff
@@ -92,9 +95,7 @@ function TChannel(options) {
         self.logger.warn('server socket close');
     });
 
-    if (self.listening) {
```

Comment:
can we please leave the flag until we remove the .listen() call itself from the constructor?